### PR TITLE
Note the huge prefix/null byte gotcha in README

### LINKED ISSUE #36: please modify example in README
key >= "050"
-->
bytes.Compare(key, []byte("050"))

### LINKED ISSUE #37: SeekLowerBound not working correctly
I was trying to replace our custom `Seek` behavior in my fork with `SeekLowerBound` described in here https://github.com/hashicorp/go-immutable-radix/pull/29#issuecomment-627942830 but hit some issues.

First thing I discovered was that an empty key (root) confuses the seeker. Tried to fix this in https://github.com/hashicorp/go-immutable-radix/commit/30c53379998b3a3fc23b26088546e455d5cd25f4

But then I discovered some more cases that behave weirdly. Added tests for them in https://github.com/hashicorp/go-immutable-radix/commit/5bed3aa0fa9f25207cbbe490ebc97ce18aeae6d2

```
    --- FAIL: TestIterateLowerBound/case015 (0.00s)
        iradix_test.go:1712: keys=[ aaa bbb] search=aa want=[aaa bbb]
        iradix_test.go:1730: mis-match: key=aa
              got=[]
              want=[aaa bbb]
    --- FAIL: TestIterateLowerBound/case016 (0.00s)
        iradix_test.go:1712: keys=[ aaa bbb] search=aaab want=[bbb]
        iradix_test.go:1730: mis-match: key=aaab
              got=[]
              want=[bbb]
    --- PASS: TestIterateLowerBound/case017 (0.00s)
        iradix_test.go:1712: keys=[ aaa bbb] search= want=[ aaa bbb]
    --- FAIL: TestIterateLowerBound/case018 (0.00s)
        iradix_test.go:1712: keys=[a aa bbb] search=aa want=[aa bbb]
        iradix_test.go:1730: mis-match: key=aa
              got=[bbb]
              want=[aa bbb]
    --- FAIL: TestIterateLowerBound/case019 (0.00s)
        iradix_test.go:1712: keys=[a aaa bbb] search=aa want=[aaa bbb]
        iradix_test.go:1730: mis-match: key=aa
              got=[bbb]
              want=[aaa bbb]
```

@banks @thaJeztah

### PR DESCRIPTION
**Pull request by banks**
Fixes #37

Also fixes #36

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,6 +15,12 @@ in a more efficient manner than performing each operation one at a time.
 
 For a mutable variant, see [go-radix](https://github.com/armon/go-radix).
 
+## :warning: This tree implementation is low-level and can be easily used incorrectly
+   - **No key may be a prefix of any other.**
+     - This is _not_ enforced but _will cause undefined behavior_ if violated. 
+     - The typical way to work around this is to append a null byte (`0x0`) to each key unless you know in advance your keys will never be prefixes. We don't always do this in tests if we arranged them to not be prefixes anyway!
+     - We could avoid this gotcha by automatically adding null bytes to all keys but that has high cost of causing re-allocation of keys for every operation that the caller might not want if they already.
```
Comment (ncabatoff) [inline README.md:22]: > if they already.

This sentence seems to end abruptly.

What about adding new methods e.g. `SafeInsert` that append the null byte?  So people have to opt-in, but at least new users of the lib are guided to using it in a way with less footguns?

**Comment by banks** [comment]
I'm actually having doubts about this now. The example in the readme failed this invariant already but still worked fine even if we use Get as well for both keys: ttps://play.golang.org/p/2pMAjaqbPP4

There are a few things going on here:
 - memDB assumed that null bytes are always needed in iradix though we didn't document it here
 - when I added `SeekLowerBound` I ran into this and "discovered" that at least memDB has always assumed a null byte is necessary even before.
 - but it seems that some subset of the functionality _does_ work OK even if one key is the prefix of another.

Perhaps it would be possible to implement `SeekLowerBound` differently if all other functionality doesn't have this requirement 🤔.

**Comment by banks** [comment]
Looks like I came to a bad conclusion doing the `LowerBound` work about this invariant. I don't believe any other operation on the tree does require that although memdb and one other radix implementation from the same original author I was looking at at the time both do always nul terminate and require that.

So I made `SeekLowerBound` only work when no keys are prefixes of each other despite the rest of the lib aparently not having that constraint - the README example shows it! Tests for other operations clearly expect that to work.

I'm going to look at whether it's possible to make `SeekLowerBound` work for this case too to "fix" the issue and then this notice will be redundany. If it's not for some reason I will update this PR to note that this only matters if you use lower/upper bounds.

**Comment by banks** [comment]
Ha ha this is totally wrong! I'm glad about that 😄 

See #39

**Pull request closed without merging**